.gitignore

```diff
@@ -1 +1,2 @@
 .env.local
+postgres/storage/*
```

Comment:
Already using a volume. This is mainly for git to not track the database files that are created. The same way `synapse/data` directory has a local `.gitignore` entry for it :)

Reply:
Indeed it's currently already in a volume, but it's a volume that stores the files on the host inside of the source directory. With my example above, the files would still be stored on the host, but not on the source directory, so you wouldn't need to add them to `.gitignore`.

Since these files aren't useful (i.e. you will probably never open them directly with an editor), I think it would be better to remove them from the source directory, so that, for example, IDEs don't waste resources indexing the files in it.

It's just a minor detail, feel free to ignore.

Reply:
@psrpinto I do want to understand your point better and get it right. My intention here is to have the db files outside so that destroying the container don't destroy the database state and a quick way to move or restore the database, if ever needed. I hope that explains why I have gone this route.

Now, that being said, would you say, having it this way is fine or some other way is better? Perhaps I should call `storage` dir as `data` for consistency? Open to anything :)

Reply:
I agree storing the database data outside the postgres container is the way to go. So that means storing it in a docker volume, which is what you did, and also what I was suggesting :) I think maybe I didn't explain myself correctly.

AFAICT there's two ways to define a docker volume:

1. You tell docker to turn directory `/foo/bar` (or just `./bar`) into a volume
2. You tell docker to create a "named volume". Docker will store the files in a directory where it places the data of all named volumes. You don't know where this directory is, and you also don't care, docker manages it for you.

Independently of how the volume was created, the volume is then mounted into a path in the container.

What I was suggesting was to go with option 2.:

```yml
services:
  postgres:
    image: postgres:14
    volumes:
      - postgres-data:/var/lib/postgresql/data

volumes:
  postgres-data:
```

Reply:
Gotcha! I understand it now. I would like to keep option 1 here, as I do care about knowing the exact directory for easy backup & restore when experimenting.